storagenode/piecestore/endpoint.go

```diff
@@ -181,6 +182,28 @@ func (endpoint *Endpoint) DeletePieces(
 	}, nil
 }
 
+func (endpoint *Endpoint) flagUnusualSlowUpload(averageUploadRate float64, connectionCount int64) bool {
```

Comment:
nit: Let's move this after `Upload` -- i.e. less important funcs below the important ones.

Reply:
nit: currently the method name is a verb, which would indicate that it changes something... also the name isn't super clear on what the "true/false" means... maybe just `isSlowUpload` or `verifySlowUpload(...) error` to return the appropriate error?

Reply:
Indeed, looks better that way